mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMailboxMapper.java

```diff
@@ -129,27 +131,24 @@ private JPAMailbox jpaMailbox(Mailbox mailbox) throws MailboxException {
     }
 
     private boolean isPathAlreadyUsedByAnotherMailbox(MailboxPath mailboxPath) throws MailboxException {
-        try {
-            findMailboxByPath(mailboxPath);
-            return true;
-        } catch (MailboxNotFoundException e) {
-            return false;
-        }
+        return findMailboxByPath(mailboxPath)
+            .blockOptional()
+            .isPresent();
     }
 
     @Override
-    public Mailbox findMailboxByPath(MailboxPath mailboxPath) throws MailboxException, MailboxNotFoundException {
+    public Mono<Mailbox> findMailboxByPath(MailboxPath mailboxPath)  {
```

Comment:
can this method be done in a reactive fashion?

Reply:
No I don't think so.

I'd rather keep the behaviour unchanged here...